### ISSUE TITLE
Getting feature categories from model's `feature_dtypes_` field causes key error

### ISSUE THREAD

**Issue by EmilyMohellenbrandQC** [Quantco/glum#735]
The `SpoxGLMRegressor` gets feature categories from its model's `feature_dtypes_` field, see [here](https://github.com/Quantco/quantcore.learn/blob/ff66feff4f2edd4711fd06e2d9c0285deb697aa7/quantcore/learn/onnx/_spox_converters/_glm_regressor.py#L113C42-L113C42). 

The problem is `feature_dtypes_` might contain features that are not in use. In such case, unused features might not exist in input `X`, then it would cause KeyError in line 105 or 107.

Solution: `feature_names_` seems to contain only the used features.

**Comment by MarcAntoineSchmidtQC** [comment]
If I remember correctly, @lbittarello added this attribute specifically for Spox. Looking quickly at the code, my understanding is that `feature_dtypes_` is not expanded when we have categoricals. Is that the problem you are facing?

A minimal example of the problem would be helpful here. I'm not sure how "unused" features are getting in the data currently, so I need to understand what you mean here.

**Comment by EmilyMohellenbrandQC** [comment]
The code I had problem with was client code, which could not be copied here. But the simple description is following: 
- I'm trying to convert a model to ONNX. The model was handed to me. I did not develop this model so I don't know the internal details. 
- I got `KeyError` from `X[fname]` emitted by [line](https://github.com/Quantco/quantcore.learn/blob/ff66feff4f2edd4711fd06e2d9c0285deb697aa7/quantcore/learn/onnx/_spox_converters/_glm_regressor.py#L107) ... 
- Debug shows that `fname` is a feature name that comes from `self.model.feature_dtypes_` see [line](https://github.com/Quantco/quantcore.learn/blob/ff66feff4f2edd4711fd06e2d9c0285deb697aa7/quantcore/learn/onnx/_spox_converters/_glm_regressor.py#L113C24-L113C24), but it doesn't exist in `X`.  
- Checking with the modeling team, I was told that this feature is no longer relevant (we used to use this feature, but not any more). Somehow this no longer in use feature got in `feature_dtypes_`. So we got to the conclusion that `feature_dtype_` includes all (used and unused) features. Again, I did not develop the model, so I can't tell the details.
- I did a comparison of `self.model.feature_names_` and `self.model.feature_dtypes_`, their length is not the same. `self.model.feature_names_` is missing exactly that no longer in use feature. So I come to the conclusion that `feature_names_` might be the proper source to get features.
